smithy-trait-codegen/src/test/resources/META-INF/smithy/non-java-names.smithy

```diff
@@ -0,0 +1,15 @@
+$version: "2.0"
+
+namespace test.smithy.traitcodegen.names
+
+/// Snake cased
+@trait
+structure snake_case_structure {
```

Comment:
I dont think this case realistically something we should try to deconflict. Im not sure it would work in most of our other code generators and I think we should probably fail on this kind of naming conflict. We could add a validator to check if we think this is likely.